### ISSUE TITLE
Blog Posts block: Make featured images clickable, and linked to posts.

### ISSUE THREAD

**Issue by happychait** [Automattic/wp-calypso#75597]
### What

On a Blog Posts block, make the featured images clickable, and linked to the individual blog posts.

### Why

A customer requested this feature in 42499329-hc

On testing, I found that featured images on the Blog Posts block are not clickable.

Featured images on Query Loop Block and Post Carousel blocks are clickable and linked to individual posts. It would be nice for the Blog Posts block to have that feature too.

**Comment by cuemarie** [comment]
Hey @happychait ! 

I ran a quick test on a simple site using Zoologist, and I see that the Featured Image is linked to the post itself, take a look:


https://user-images.githubusercontent.com/27249804/231542959-b8a449fc-899a-4c22-9d97-0ff2bf54596e.mp4


Are you seeing that not work this way in your tests?  I wonder if there might be a discrepancy between certain themes, perhaps?

**Comment by happychait** [comment]
I am unsure what's changed between when I reported this and now, but you are right! 

I tested this again on two separate test sites and can't reproduce this anymore.

I swear the featured images weren't clickable when I was in the chat with the customer, but I can no longer reproduce this on Blank Canvas ([in this test post](https://chaitat.com/blog-posts-block-with-featured-images/)) and Heiwa theme ([in this test post](https://chaitpro.blog/blog-posts-block-with-featured-images/))

So, closing this report.
